### PR TITLE
MESOS: Fix daemonsets on Mesos

### LINKED ISSUE #502: e2e:  Kubernetes e2e suite.Daemon set should launch a daemon pod on every node of the cluster
This is about an experimental features. It probably should not be part of conformance tests:
```
Failure
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/daemon_set.go:49
Expected error:
    <*errors.errorString | 0xc2080f0a60>: {
        s: "timed out waiting for the condition",
    }
    timed out waiting for the condition
not to have occurred
```

### PR DESCRIPTION
**Pull request by sttts**
This PR enables the support for daemonsets in the apiserver such that the daemonset e2e test is passed. This test is part of the conformance test suite.

Prerequisites:
- NodeLabels support: https://github.com/kubernetes/kubernetes/pull/13857
- daemonset bug fixes: https://github.com/kubernetes/kubernetes/pull/14550

=> only the last two commits are relevant.

Fixes https://github.com/mesosphere/kubernetes-mesos/issues/502

### CODE REVIEW

**Comment by sttts** [comment]
I think this is much easier to read. And about the propositional logic itself, there is not much to improve.

**Comment by sttts** [comment]
The condition will even appear in the function if we factor that out.

**Comment by s-urbaniak** [comment]
ok, convinced ... LGTM :-)

File: cluster/mesos/docker/docker-compose.yml [hunk 1]
```diff
@@ -85,6 +85,7 @@ apiserver:
     --cloud-config=/opt/mesos-cloud.conf
     --tls-cert-file=/var/run/kubernetes/auth/apiserver.crt
     --tls-private-key-file=/var/run/kubernetes/auth/apiserver.key
+    --runtime-config=experimental/v1alpha1
```
Comment (karlkfi) [inline cluster/mesos/docker/docker-compose.yml]: nice! glad it's this easy.

Comment (sttts) [inline cluster/mesos/docker/docker-compose.yml]: yeah, plus the few upstream fixes. but luckily it just works (tm) then.

**Comment by jdef** [comment]
this can be rebased yes? now that the daemonset fixes and node labels have landed upstream?

**Comment by sttts** [comment]
Right. Will do.

**Comment by sttts** [comment]
Rebased.

File: contrib/mesos/pkg/executor/executor.go [hunk 2]
```diff
@@ -376,21 +377,22 @@ func (k *KubernetesExecutor) handleChangedApiserverPod(pod *api.Pod) {
...
-		update := kubelet.PodUpdate{
-			Op:   kubelet.UPDATE,
-			Pods: []*api.Pod{oldPod},
+	if oldPod != nil && pod.Status.Phase == api.PodRunning {
+		timeModified := differentTime(oldPod.DeletionTimestamp, pod.DeletionTimestamp)
+		graceModified := differentPeriod(oldPod.DeletionGracePeriodSeconds, pod.DeletionGracePeriodSeconds)
+		if timeModified || graceModified {
+			log.Infof("pod %s/%s is terminating at %v with %vs grace period, telling kubelet", pod.Namespace, pod.Name, *pod.DeletionTimestamp, *pod.DeletionGracePeriodSeconds)
+
+			// modify pod in our registry to avoid that other changed bleed into the kubelet
```
Comment (jdef) [inline contrib/mesos/pkg/executor/executor.go]: would you mind clarifying "to avoid that other changed bleed into the kubelet"?

**Comment by jdef** [comment]
nit, otherwise lgtm

File: contrib/mesos/pkg/executor/executor.go (hunk 2 above)
Comment (sttts) [inline contrib/mesos/pkg/executor/executor.go]: done

**Comment by sttts** [comment]
Rebased & squashed

**Comment by sttts** [comment]
@k8s-bot test this

**Comment by sttts** [comment]
Rebased.

**Pull request merged**